File: 0830-positions-of-large-groups/0830-positions-of-large-groups.py
class Solution:
    def largeGroupPositions(self, s: str) -> List[List[int]]:
        
        
        n = len(s)
        res = []

        if(n < 3):
            return res
        
        i=0
        for r in range(n):
            if r == n-1 or s[r] != s[r+1]:
                if r-i+1 >= 3:
                    res.append([i, r])
                
                i=r+1
        
        return res

File: 0830-positions-of-large-groups/0830-positions-of-large-groups.java
class Solution {
    public List<List<Integer>> largeGroupPositions(String s) {
        
        
        List<List<Integer>> res = new ArrayList<>();
        
        int i = 0;
        
        for(int r=0;r<s.length();r++){
            if(r == s.length()-1 || s.charAt(r) != s.charAt(r+1)){
                if(r-i+1 >= 3){
                    res.add(List.of(i, r));
                }
                
                i = r+1;
            }
        }
        return res;
    }
}